File: test/resources/repos/cpp/test_repo/ConcreteProcessor.cpp
#include "ConcreteProcessor.h"
#include <iostream>

ConcreteProcessor::ConcreteProcessor(const std::string& name, const std::string& type) 
    : BaseClass(name), type(type) {}

void ConcreteProcessor::process() {
    std::cout << "Processing " << name << std::endl;
}

void ConcreteProcessor::execute() {
    process();
}

std::string ConcreteProcessor::getType() const {
    return type;
}

File: test/resources/repos/typescript/test_repo/ConcreteProcessor.ts
import { BaseClass, Processable } from './BaseClass';

export class ConcreteProcessor extends BaseClass implements Processable {
    private data: string[];

    constructor(name: string) {
        super(name);
        this.data = [];
    }

    execute(): void {
        console.log(`Executing processor ${this.name}`);
        this.process();
    }

    process(): void {
        console.log(`Processing data: ${this.data.join(', ')}`);
    }

    getType(): string {
        return 'ConcreteProcessor';
    }

    addData(item: string): void {
        this.data.push(item);
    }
}